packages/react-data-grid/src/HeaderRow.js

```diff
@@ -83,8 +83,22 @@ const HeaderRow = createReactClass({
   },
 
   getSortableHeaderCell(column) {
-    let sortDirection = (this.props.sortColumn === column.key) ? this.props.sortDirection : SortableHeaderCell.DEFINE_SORT.NONE;
-    return <SortableHeaderCell columnKey={column.key} onSort={this.props.onSort} sortDirection={sortDirection}/>;
+    let SortableRenderer = SortableHeaderCell;
+    if (column.headerRenderer !== undefined) {
+      SortableRenderer = column.headerRenderer;
+    }
+    const sortDirection = (this.props.sortColumn === column.key) ? this.props.sortDirection : SortableHeaderCell.DEFINE_SORT.NONE;
+    const sortDescendingFirst = (column.sortDescendingFirst === undefined ) ? false : column.sortDescendingFirst;
+    const props = { columnKey: column.key, onSort: this.props.onSort, sortDirection: sortDirection, sortDescendingFirst };
+
+    if (React.isValidElement(SortableRenderer)) {
```

Comment:
Can you catch your branch up with the master branch again please? This particular change came from another pull request that had to be reverted due to an issue, so we dont want this going back into master